python/lsst/the/monster/measure_synth_colorterms.py

```diff
@@ -0,0 +1,268 @@
+                template_lambda = templates[i]['lambda']
+                template_f_lambda = templates[i]['flux']
+                template_f_nu = template_f_lambda * template_lambda * template_lambda
+
+                int_func = interpolate.interp1d(template_lambda, template_f_nu)
+                tput_lambda = throughputs[band]["wavelength"]*10.
+                f_nu = np.zeros(tput_lambda.size)
+                # Make sure we interpolate in range
+                good, = np.where((tput_lambda >= template_lambda[0]) & (tput_lambda <= template_lambda[-1]))
+                f_nu[good] = int_func(tput_lambda[good])
+                # out of range, let it hit the limit
+                lo, = np.where(tput_lambda < template_lambda[0])
+                f_nu[lo] = int_func(tput_lambda[good[0]])
+                hi, = np.where(tput_lambda > template_lambda[-1])
+                f_nu[hi] = int_func(tput_lambda[good[-1]])
+
+                num = integrate.simpson(f_nu*throughputs[band]["throughput"]/tput_lambda, tput_lambda)
+                denom = integrate.simpson(throughputs[band]["throughput"]/tput_lambda, tput_lambda)
+
+                synth_lsst_cat[lsst_info.get_flux_field(band)][i] = num/denom
+
+        # Now the DES catalog
+        des_passbands = astropy.table.Table.read(self.des_throughput_file(), format="fits")
+
+        # Patch in the y -> Y
+        ind = des_info.bands.index("y")
+        des_info.bands[ind] = "Y"
+
+        dtype = []
+        for band in des_info.bands:
+            flux_field = des_info.get_flux_field(band)
+            dtype.append((flux_field, "f8"))
+            dtype.append((flux_field + "Err", "f8"))
+
+        synth_des_cat = np.zeros(n_templates, dtype=dtype)
+
+        for i in range(n_templates):
+            for j, band in enumerate(des_info.bands):
+                template_lambda = templates[i]['lambda']
+                template_f_lambda = templates[i]['flux']
+                template_f_nu = template_f_lambda * template_lambda * template_lambda
+
+                int_func = interpolate.interp1d(template_lambda, template_f_nu)
+                tput_lambda = des_passbands["LAMBDA"]
+                f_nu = np.zeros(tput_lambda.size)
+                # Make sure we interpolate in range
+                good, = np.where((tput_lambda >= template_lambda[0]) & (tput_lambda <= template_lambda[-1]))
+                f_nu[good] = int_func(tput_lambda[good])
+                # out of range, let it hit the limit
+                lo, = np.where(tput_lambda < template_lambda[0])
+                f_nu[lo] = int_func(tput_lambda[good[0]])
+                hi, = np.where(tput_lambda > template_lambda[-1])
+                f_nu[hi] = int_func(tput_lambda[good[-1]])
+
+                num = integrate.simpson(f_nu*des_passbands[band]/tput_lambda, tput_lambda)
+                denom = integrate.simpson(des_passbands[band], tput_lambda)
+
+                synth_des_cat[des_info.get_flux_field(band)][i] = num/denom
```

Comment:
would it be better to convert this loop and the one on L132 to a method instead of duplicate code? I dont feel strongly here.

Reply:
There were subtle differences (the units) that I just didn't want to deal with it, so I'm leaving it as it is.  However, it's a lot shorter now that I learned I can set two fill values!